source/agora/node/Ledger.d

```
@@ -229,14 +239,63 @@ public class Ledger
         else
             return null;
     }
+
+    /***************************************************************************
+
+        Write a block to the ledger file on disk.
+
+        Params:
+            fd = Instance of LedgerFile
+            block = Type of struct to writeFile
+
+    ***************************************************************************/
+
+    public void writeBlockFile (File file, ref Block block) @trusted
+    {
+        const string dir = this.data_dir;
+        size_t block_size = 0;
+
+         // sure we're in the right directory
+        assert(dir.exists && std.file.isDir(dir));
+
+        scope SerializeDg dg = (scope const (ubyte[]) data) nothrow @safe
+        {
+            try
+            {
+                file.rawWrite(data);
+                block_size += data.length;
+            }
+            catch (Exception ex)
+            {
+                //logError("writeBlockFile: %s", ex.message);
```

Errors should never be silently ignored